From f0fa7254fb1326489daa88ffed6af6d48cad49d0 Mon Sep 17 00:00:00 2001
From: fujiwarat <takao.fujiwara1@gmail.com>
Date: Thu, 11 Oct 2012 16:04:38 +0900
Subject: [PATCH] Fix typo in data/dconf/profile/ibus

http://git.gnome.org/browse/gdm/commit/?id=aa25b559c6e907485cc9e
I copied the typo by mitake. dconf_engine_source_new() supports
"user-db" or "system-db".
http://git.gnome.org/browse/dconf/tree/engine/dconf-engine-source.c#n75
---
 data/dconf/profile/ibus |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/dconf/profile/ibus b/data/dconf/profile/ibus
index 1d3fc5f..aa8fe5f 100644
--- a/data/dconf/profile/ibus
+++ b/data/dconf/profile/ibus
@@ -1,2 +1,2 @@
 user-db:user
-systemd-db:ibus
+system-db:ibus
-- 
1.7.10.4

From 065ad6c44a665526796dffe91ac7c2ebd83ecd71 Mon Sep 17 00:00:00 2001
From: fujiwarat <takao.fujiwara1@gmail.com>
Date: Thu, 11 Oct 2012 16:43:06 +0900
Subject: [PATCH] Fix fallback 'C' locale in ibus-setup.

---
 setup/main.py |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/setup/main.py b/setup/main.py
index 8a2391d..81e4301 100644
--- a/setup/main.py
+++ b/setup/main.py
@@ -548,7 +548,11 @@ class Setup(object):
         Gtk.main()
 
 if __name__ == "__main__":
-    locale.setlocale(locale.LC_ALL, '')
+    try:
+        locale.setlocale(locale.LC_ALL, '')
+    except locale.Error:
+        print >> sys.stderr, "IBUS-WARNING **: Using the fallback 'C' locale"
+        locale.setlocale(locale.LC_ALL, 'C')
     i18n_init()
     setup = Setup()
     setup.run()
-- 
1.7.10.4

